sound/soc/codecs/rt700.c

```diff
@@ -1339,7 +1326,33 @@ static int rt700_pcm_hw_params(struct snd_pcm_substream *substream,
 	snd_soc_component_write(component, RT700_ADC_FORMAT_L, val);
 
 	dev_err(dai->dev, "5 %s %s", __func__, dai->name);
-	return retval;
+	return 0;
+}
+
+static int rt700_prepare_stream(struct snd_pcm_substream *substream,
+				struct snd_soc_dai *dai)
+{
+	struct snd_soc_component *component = dai->component;
+	struct rt700_priv *rt700 = snd_soc_component_get_drvdata(component);
+	struct sdw_stream_data *stream;
+	int retval;
+
+	stream = snd_soc_dai_get_dma_data(dai, substream);
+
+	if (!stream) {
+		dev_err(dai->dev, "no dma data found in dai %s", dai->name);
+		return -ENOMEM;
+	}
+
+	retval = sdw_stream_add_slave(rt700->slave, &rt700->stream_config,
+				      &rt700->port_config, 1,
+				      stream->sdw_stream);
```

Comment:
I don't fully get this code, is this going to initialize the 'stream'?
Again please add comments to explain what you are doing, the code is not self-explanatory.

Reply:
@RanderWang yeah same here. I dont understand why we need to move this to prepare. AFAICT, sdw_stream_add_slave() need the stream_config to be set, particularly the frame_rate.